plugins/elasticsearch/encoders.go

```diff
@@ -88,12 +103,17 @@ func writeQuotedString(b *bytes.Buffer, str string) {
 
 }
 
-func writeField(first bool, b *bytes.Buffer, f *message.Field, raw bool) {
+func writeField(first bool, b *bytes.Buffer, f *message.Field, raw bool, replaceDotsWith string) {
 	if !first {
 		b.WriteString(`,`)
 	}
 
-	writeQuotedString(b, f.GetName())
+        if replaceDotsWith != "." {
+	    writeQuotedString(b, replaceDots(f.GetName(), replaceDotsWith))
```

Comment:
Not sure what the value of replaceDotsWith function is over the stdlib's strings.Replace:

`strings.Replace(f.GetName(), ".", replaceDotsWith, -1)`